### PR TITLE
fix: Fenix does not require WRITE_EXTERNAL_STORAGE permission to be granted

### LINKED ISSUE #2304: Can not grant android.permission.WRITE_EXTERNAL_STORAGE to org.mozilla.fenix
### Is this a feature request or a bug?
Bug

### What is the current behavior?
Can't test my extensions on Firefox for Android because it's package doesn't ask for android.permission.WRITE_EXTERNAL_STORAGE.
```
D:\MyUser\Dev\Vs\PrivateCookies>web-ext run -t firefox-android --adb-device ifjbnnvsmv7hbqwg --firefox-apk org.mozilla.fenix
Running web extension from D:\MyUser\Dev\Vs\PrivateCookies
Selected ADB device: ifjbnnvsmv7hbqwg

s: Required android.permission.WRITE_EXTERNAL_STORAGE has not be granted for org.mozilla.fenix. Please grant them using the Android Settings or using the following adb command:
         adb shell pm grant org.mozilla.fenix android.permission.WRITE_EXTERNAL_STORAGE



D:\MyUser\Dev\Vs\PrivateCookies>adb shell pm grant org.mozilla.fenix android.permission.WRITE_EXTERNAL_STORAGE

Exception occurred while executing 'grant':
java.lang.SecurityException: Package org.mozilla.fenix has not requested permission android.permission.WRITE_EXTERNAL_STORAGE
        at com.android.server.pm.permission.BasePermission.enforceDeclaredUsedAndRuntimeOrDevelopment(BasePermission.java:446)
        at com.android.server.pm.permission.PermissionManagerService.grantRuntimePermissionInternal(PermissionManagerService.java:1491)
        at com.android.server.pm.permission.PermissionManagerService.grantRuntimePermission(PermissionManagerService.java:1444)
        at com.android.server.pm.PackageManagerShellCommand.runGrantRevokePermission(PackageManagerShellCommand.java:2300)
        at com.android.server.pm.PackageManagerShellCommand.onCommand(PackageManagerShellCommand.java:251)
        at android.os.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:98)
        at android.os.ShellCommand.exec(ShellCommand.java:44)
        at com.android.server.pm.PackageManagerService.onShellCommand(PackageManagerService.java:22202)
        at android.os.Binder.shellCommand(Binder.java:929)
        at android.os.Binder.onTransact(Binder.java:813)
        at android.content.pm.IPackageManager$Stub.onTransact(IPackageManager.java:4687)
        at com.android.server.pm.PackageManagerService.onTransact(PackageManagerService.java:4449)
        at android.os.Binder.execTransactInternal(Binder.java:1159)
        at android.os.Binder.execTransact(Binder.java:1123)
```

### What is the expected or desired behavior?
The extension should be temporarily loaded in Firefox for Android.

### Version information (for bug reports)

* **Firefox version**: Nightly 93.0a1
* **OS and version**: Android 11 RP1A.200720.011 (MIUI 12.5.2.0 RGGEUXM)

````
node version: v14.17.0
npm version: 6.14.13
web-ext version: 6.3.0
````

### PR DESCRIPTION
**Pull request by rpl**
Fixes #2304

For Fenix we don't need the WRITE_EXTERNAL_STORAGE permission to be granted:
- it was required for Fennec because `web-ext run` was also running it in a separate temporary profile as we do on desktop
- but Fennec doesn't support that and so the WRITE_EXTERNAL_STORAGE permission isn't really necessary
- We still need the READ_EXTERNAL_STORAGE permission to have been granted, to allow Fennec to load the xpi file that `web-ext run` uploads to the android emulator or device

TODO:
- [x] confirm with Rob that we are both ok with the strategy drafted in this PR (or agree on an alternative one)
- [x] ~~add a more specific test case for the behavior expected~~
- [x] test on an android 11 emulator or device and a recent Fenix and GeckoViewExample builds

### CODE REVIEW

File: src/extension-runners/firefox-android.js [hunk 1]
```diff
@@ -411,17 +411,34 @@ export class FirefoxAndroidExtensionRunner {
...
     const requiredPermissions = [
       'android.permission.READ_EXTERNAL_STORAGE',
-      'android.permission.WRITE_EXTERNAL_STORAGE',
     ];
 
     await adbUtils.ensureRequiredAPKRuntimePermissions(
       selectedAdbDevice, selectedFirefoxApk, requiredPermissions
     );
+
+    // Runtime permission needed to be able to run Firefox on a temporarily created profile
```
[the remark this answers is not in the document]
Comment (rpl) [reply src/extension-runners/firefox-android.js:424]: > Is this really needed? The profile directory is now created in `/data/local/tmp/` instead of `/sdcard/`: https://github.com/mozilla/web-ext/blame/c884696fd3ea496c67959337f98d4e39bdb08871/src/util/adb.js#L14

I know that ;-), and I also described it explicitly in the pull request description.

Personally I'm not really against removing it all together, the only reason I left it as a debug message is that we still have docs related to Fennec (e.g. [in extensionworkshop.com](https://extensionworkshop.com/documentation/develop/developing-extensions-for-firefox-for-android-fennec/), as you already know), but even removing it wouldn't prefer a user to use web-ext with Fennec (the extension developer would need to be sure to add both those prefs).

I'll update the PR to remove it (it may also be enough to bring back the coverage to the previous threshold and make codecov happy again on this PR).

File: src/extension-runners/firefox-android.js [hunk 2]
```diff
@@ -411,12 +411,10 @@ export class FirefoxAndroidExtensionRunner {
     log.debug('Checking read/write permissions needed for web-ext' +
               `on ${selectedFirefoxApk}...`);
 
-    // Runtime permission needed to be able to run Firefox on a temporarily created profile
-    // on android versions >= 23 (Android Marshmallow, which is the first version where
-    // these permissions are optional and have to be granted explicitly).
+    // Runtime permissions needed to Firefox to be able to access the
+    // xpi file uploaded to the android device or emulator.
     const requiredPermissions = [
       'android.permission.READ_EXTERNAL_STORAGE',
```
Comment (Rob--W) [inline src/extension-runners/firefox-android.js:417]: I just tested, and the read permission isn't even needed to run an extension in Fenix. We can delete lots of code.

**Review by Rob--W — CHANGES_REQUESTED** [review]

File: src/extension-runners/firefox-android.js (hunk 2 above)
Comment (rpl) [reply src/extension-runners/firefox-android.js:417]: @Rob--W have you checked that the permission wasn't actually already granted?
some more details about what STR you actually tested manually would be helpful to assess if that was enough to ensure we are not missing some legit corner case.

Comment (rpl) [reply src/extension-runners/firefox-android.js:417]: @Rob--W also, I would like to release this into a minor or patch version, and so "less changes" would make me feel more less concerned than "remove a lot of code" about releasing it in a "non major" new release.

I would be obviously more than happy about "remove a lot of code" in a separate follow up.

Comment (Rob--W) [reply src/extension-runners/firefox-android.js:417]: Filed https://github.com/mozilla/web-ext/issues/2312
